File: day-6/main.py
startMarkerLength = 4
endMarkerLength = 14

start = False
end = False

with open('input', 'r') as file:
    stream = file.readline()
    for i in range(startMarkerLength - 1, len(stream)):
        # Find start marker
        if not start and len(set(stream[i - startMarkerLength:i])) == startMarkerLength:
            start = i

        # Find end marker
        if not end and i > endMarkerLength and len(set(stream[i - endMarkerLength:i])) == endMarkerLength:
            end = i
            break


print('Part 1: {}'.format(start))
print('Part 2: {}'.format(end))

File: day-6/main.ts
const startMarkerLength: number = 4;
const endMarkerLength: number = 14;

let start: number | false = false;
let end: number | false = false;

const fs = require('fs');
const stream: string = fs.readFileSync('input', 'utf-8');

for (let i = startMarkerLength - 1; i < stream.length; i++) {
  // Find start marker
  if (!start && new Set(stream.slice(i - startMarkerLength + 1, i + 1)).size === startMarkerLength) {
    start = i;
  }

  // Find end marker
  if (!end && i > endMarkerLength && new Set(stream.slice(i - endMarkerLength + 1, i + 1)).size === endMarkerLength) {
    end = i;
    break;
  }
}

console.log(`Part 1: ${start}`);
console.log(`Part 2: ${end}`);
